src/engine/routing_algorithms/shortest_path.cpp

```diff
@@ -453,56 +457,55 @@ shortestPathSearchImpl(SearchEngineData &engine_working_data,
                  total_weight_to_reverse != INVALID_EDGE_WEIGHT);
 
     // We make sure the fastest route is always in packed_legs_to_forward
-    if (total_weight_to_forward > total_weight_to_reverse)
+    if (total_weight_to_forward < total_weight_to_reverse ||
+        (total_weight_to_forward == total_weight_to_reverse &&
+         total_packed_path_to_forward.size() < total_packed_path_to_reverse.size()))
```

Comment:
Can be written as `std::tie(total_weight_to_forward, total_packed_path_to_forward.size()) < std::tie(total_weight_to_reverse, total_packed_path_to_reverse.size())`